### PR TITLE
Issue 353 race condition

### PR DESCRIPTION
**Pull request by 2infinitee**
@integrii I think I am ready for review and your take on how I am coding this.

### CODE REVIEW

File: pkg/checks/external/main.go [hunk 1]
```diff
@@ -339,6 +339,21 @@ func (ext *Checker) setUUID(uuid string) error {
...
+	// uuid before it has been update on kh server, the pod will error out. This for-loop is to give it some wait time
+	// before it goes to check with kuberhealthy.
+	waitDuration = 0
+	for {
+		time.Sleep(time.Second * 3)
+		if checkState.Spec.CurrentUUID = ext.currentCheckUUID {
+			break
+		}
+		if waitDuration = 3 {
+			break
```
[the remark this answers is not in the document]
Comment (2infinitee) [reply pkg/checks/external/main.go:351]: got some suggestion will commit my changes

File: pkg/checks/external/main.go [hunk 2]
```diff
@@ -339,6 +339,21 @@ func (ext *Checker) setUUID(uuid string) error {
 		_, err = ext.KHStateClient.Update(checkState, stateCRDResource, ext.Name(), ext.CheckNamespace())
 	}
 
+	// Sometimes a race condition occurs when a pod has to verify uuid with kh server. If the pod happens to check the
+	// uuid before it has been update on kh server, the pod will error out. This for-loop is to give it some wait time
+	// before it goes to check with kuberhealthy.
+	waitDuration = 0
```
[the remark this answers is not in the document]
Comment (2infinitee) [reply pkg/checks/external/main.go:344]: changed waitDuration to tries and also fixed operators

File: pkg/checks/external/main.go [hunk 3]
```diff
@@ -339,6 +339,21 @@ func (ext *Checker) setUUID(uuid string) error {
...
+	for {
+		time.Sleep(time.Second * 3)
+		if checkState.Spec.CurrentUUID = ext.currentCheckUUID {
+			break
+		}
+		if waitDuration = 3 {
```
[the remark this answers is not in the document]
Comment (2infinitee) [reply pkg/checks/external/main.go:350]: will commit changes to fix operators

**Comment by 2infinitee** [comment]
@integrii added your suggestions but let me know if I'm on the right track

**Comment by joshulyne** [comment]
tested locally and it looks like its working:
```time="2020-04-14T23:55:33Z" level=info msg="85031e8e-d6e4-4c07-8d8d-8db13a61b137 kuberhealthy/daemonset: [Waiting 1 second before checking daemonset uuid.]"
time="2020-04-14T23:55:33Z" level=info msg="0de71a52-af14-49ad-b3f4-3ccd34004226 kuberhealthy/deployment: [Waiting 1 second before checking deployment uuid.]"
time="2020-04-14T23:55:33Z" level=info msg="35ad8568-1f2b-439f-a55b-0b2aa8515ac6 kuberhealthy/dns-status-internal: [Waiting 1 second before checking dns-status-internal uuid.]"```

**Comment by 2infinitee** [comment]
@joshulyne @jonnydawg @integrii branch looks good. I re-added the log when uuid matches and got help from Josh for helping me with my rebase fiasco.

**Pull request merged**